litert/vendors/qualcomm/qnn_manager.cc

```diff
@@ -300,27 +304,42 @@ LiteRtStatus QnnManager::Init(absl::Span<const QnnBackend_Config_t*> configs,
     return kLiteRtStatusErrorRuntimeFailure;
   }
 
-  if (soc_model.has_value()) {
-    soc_model_ = *soc_model;
+  if (soc_info.has_value()) {
+    soc_info_ = *soc_info;
     LITERT_LOG(LITERT_INFO,
-               "Initializing QNN backend for device architecture %d",
-               *soc_model);
-    QnnHtpDevice_CustomConfig_t arch_custom_config = {};
-    arch_custom_config.option = QNN_HTP_DEVICE_CONFIG_OPTION_ARCH;
-    arch_custom_config.arch.arch = *soc_model;
-    arch_custom_config.arch.deviceId = 0;
-
-    QnnDevice_Config_t arch_device_config = {};
-    arch_device_config.option = QNN_DEVICE_CONFIG_OPTION_CUSTOM;
-    arch_device_config.customConfig = &arch_custom_config;
-
-    const QnnDevice_Config_t* device_configs[2] = {
-        &arch_device_config,
-        nullptr,
-    };
+               "Initializing QNN backend for device architecture V%d",
+               soc_info_.dsp_arch);
+    htp_device_platform_info_config_ =
+        std::make_unique<::qnn::HtpDevicePlatformInfoConfig>();
+    htp_device_custom_config_ =
+        std::make_unique<::qnn::HtpDeviceCustomConfig>();
+    const std::vector<QnnDevice_CustomConfig_t> device_custom_config =
+        htp_device_custom_config_->CreateDeviceCustomConfig(&soc_info_);
+    const std::vector<QnnDevice_PlatformInfo_t*>& device_platform_info =
```

Comment:
I don't think we need reference here.
```suggestion
    const std::vector<QnnDevice_PlatformInfo_t*> device_platform_info =
```

Reply:
Thanks! I will fix that.